async_producer.go

```diff
@@ -13,6 +13,9 @@ import (
 	"github.com/rcrowley/go-metrics"
 )
 
+// ErrProducerRetryBufferOverflow is returned when the bridging retry buffer is full and OOM prevention needs to be applied.
+var ErrProducerRetryBufferOverflow = errors.New("discard a message and surface it as an error to prevent retry buffer to overflow")
```

Comment:
This error message does not very well explain what the error here is. I think it’s suggesting a resolution, rather than describing what’s wrong?

I would probably recommend `what went wrong: how to resolve it` as a format if we want to suggest a resolution.